servers/zts/src/main/java/com/yahoo/athenz/zts/ZTSImpl.java

```diff
@@ -1372,6 +1372,8 @@ public RoleToken getRoleToken(ResourceContext ctx, String domainName, String rol
                     ZTSConsts.ZTS_UNKNOWN_DOMAIN, principalDomain);
         }
 
+        validateNotMtlsRestricted(ctx, caller, domainName);
```

Comment:
For ZTS, we should be probably be more strict and add it to the authenticate method. We should not allow anyone with the restricted certs to connect to ZTS. I can't think of any reason, can you? This would simplify the code for any future commands as well.

Reply:
There are some methods to the ones in ZMS where user without authorization can retrieve data.
For example:
- enable users the option to check access with the restricted certs (get /access)
- Get services in a domain (GET /domain/{domainName}/service/)
- Get a service public key (GET /domain/{domainName}/service/{serviceName}/publickey/{keyId}) 
- Get a list of services to run on a specific host (GET /host/{host}/services)
- Get Status (/status)

We can block them but then why block them and not all the methods in ZMS?

Reply:
those operations can be carried out against ZMS. There is no point of going to ZTS to get a service or public keys. and if you have a restricted cert, you should not really carry out any access checks. It's basically a limited client certificate. So I think I'm ok restricting from zts server completely.